Concrete projection a hyperrectangular cartesian product

```julia
X = Interval(1, 2) × Hyperrectangle(zeros(2), ones(2))

LazySets.project(X, [1, 3])

AssertionError: package 'Polyhedra' not loaded (it is required for executing `linear_map with elimination`)
```
but the result is as hyperrectangle.

--- a/src/LazyOperations/CartesianProduct.jl
+++ b/src/LazyOperations/CartesianProduct.jl
@@ -296,7 +296,7 @@ function concretize(cp::CartesianProduct)
 end
 
 """
-    project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
+    project(cp::CartesianProduct{N, <:Interval, <:AbstractHyperrectangle{N}}, block::AbstractVector{Int}) where {N}
 
 Concrete projection of a Cartesian product between an interval and a
 hyperrectangle.
@@ -311,17 +311,17 @@ hyperrectangle.
 A set representing the projection of the cartesian product `cp` on the
 dimensions specified by `block`.
 """
-function project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
-    Δt = cp.X
+function project(cp::CartesianProduct{N, <:Interval, <:AbstractHyperrectangle{N}}, block::AbstractVector{Int}) where {N}
+    I = cp.X
     H = cp.Y
-    vars_vec = collect(vars)
+    block_vec = collect(block)
     if 1 ∉ vars
-        vars_vec .-= 1
+        block_vec .-= 1
         cH = center(H)
         rH = radius_hyperrectangle(H)
     else
-        cH = vcat(center(Δt), center(H))
-        rH = vcat(radius_hyperrectangle(Δt), radius_hyperrectangle(H))
+        cH = vcat(center(I), center(H))
+        rH = vcat(radius_hyperrectangle(I), radius_hyperrectangle(H))
     end
-    return Hyperrectangle(cH[vars_vec], rH[vars_vec], check_bounds=false)
+    return Hyperrectangle(cH[block_vec], rH[block_vec], check_bounds=false)
 end

--- a/src/LazyOperations/CartesianProduct.jl
+++ b/src/LazyOperations/CartesianProduct.jl
@@ -315,7 +315,7 @@ function project(cp::CartesianProduct{N, <:Interval, <:AbstractHyperrectangle{N}
     I = cp.X
     H = cp.Y
     block_vec = collect(block)
-    if 1 ∉ vars
+    if 1 ∉ block_vec
         block_vec .-= 1
         cH = center(H)
         rH = radius_hyperrectangle(H)

--- a/src/LazyOperations/CartesianProduct.jl
+++ b/src/LazyOperations/CartesianProduct.jl
@@ -294,3 +294,34 @@ end
 function concretize(cp::CartesianProduct)
     return cartesian_product(concretize(cp.X), concretize(cp.Y))
 end
+
+"""
+    project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
+
+Concrete projection of a cartesian product between an interval and an
+hyperrectangle.
+
+### Input
+
+- `cp`       -- cartesian product between an interval and an hyperrectangle
+- `block`    -- block structure, a vector with the dimensions of interest
+
+### Output
+
+A set representing the projection of the cartesian product `cp` on the
+dimensions specified by `block`.
+"""
+function project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
+    Δt = cp.X
+    H = cp.Y
+    vars_vec = collect(vars)
+    if 1 ∉ vars
+        vars_vec .-= 1
+        cH = center(H)
+        rH = radius_hyperrectangle(H)
+    else
+        cH = vcat(center(Δt), center(H))
+        rH = vcat(radius_hyperrectangle(Δt), radius_hyperrectangle(H))
+    end
+    return Hyperrectangle(cH[vars_vec], rH[vars_vec], check_bounds=false)
+end

--- a/src/LazyOperations/CartesianProduct.jl
+++ b/src/LazyOperations/CartesianProduct.jl
@@ -303,7 +303,7 @@ hyperrectangle.
 
 ### Input
 
-- `cp`       -- cartesian product between an interval and an hyperrectangle
+- `cp`       -- Cartesian product between an interval and a hyperrectangle
 - `block`    -- block structure, a vector with the dimensions of interest
 
 ### Output

--- a/src/LazyOperations/CartesianProduct.jl
+++ b/src/LazyOperations/CartesianProduct.jl
@@ -298,7 +298,7 @@ end
 """
     project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
 
-Concrete projection of a cartesian product between an interval and an
+Concrete projection of a Cartesian product between an interval and a
 hyperrectangle.
 
 ### Input

--- a/test/unit_CartesianProduct.jl
+++ b/test/unit_CartesianProduct.jl
@@ -385,8 +385,8 @@ for N in [Float64, Float32]
 
     # projection
     cp = LazySets.Interval(N(0), N(2)) × Hyperrectangle(N[2, 3], N[1, 1])
-    @test _project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
-    @test _project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
+    @test project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
+    @test project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
 end
 
 for N in [Float64]

--- a/test/unit_CartesianProduct.jl
+++ b/test/unit_CartesianProduct.jl
@@ -383,6 +383,10 @@ for N in [Float64, Float32]
     @test_throws AssertionError is_intersection_empty(cpa1, Universe{N}(3))
     @test_throws AssertionError is_intersection_empty(Universe{N}(5), cpa2)
 
+    # projection
+    cp = LazySets.Interval(N(0), N(2)) × Hyperrectangle(N[2, 3], N[1, 1])
+    @test _project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
+    @test _project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
 end
 
 for N in [Float64]

--- a/test/unit_CartesianProduct.jl
+++ b/test/unit_CartesianProduct.jl
@@ -384,7 +384,7 @@ for N in [Float64, Float32]
     @test_throws AssertionError is_intersection_empty(Universe{N}(5), cpa2)
 
     # projection
-    cp = LazySets.Interval(N(0), N(2)) × Hyperrectangle(N[2, 3], N[1, 1])
+    cp = Interval(N(0), N(2)) × Hyperrectangle(N[2, 3], N[1, 1])
     @test project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
     @test project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
 end

